tests/cloudpickle_test.py

```diff
@@ -2672,6 +2672,15 @@ class C:
         C1 = pickle_depickle(C, protocol=self.protocol)
         assert C1.__annotations__ == C.__annotations__
 
+    def test_class_annotations_abstractclass(self):
+        # see https://github.com/cloudpipe/cloudpickle/issues/572
+
+        class C(abc.ABC):
+            a: int
+
+        C1 = pickle_depickle(C, protocol=self.protocol)
+        assert C1.__annotations__ == C.__annotations__
```

Comment:
Please call `C2 = pickle_depickle(C1, protocol=self.protocol)` and ensure that `__annotate_func__` has been recreated and instantiate a concrete subclass of `C2` of this class in the test to make sure that the ABC machinery is still functional on the reconstructed abstract base class.